v2/api-validator/src/server/controllers/withdrawal-controller.ts

```diff
@@ -1,46 +1,164 @@
+import { XComError } from '../../error';
+import { Repository } from './repository';
+import { AssetsController } from './assets-controller';
 import {
+  BlockchainWithdrawalRequest,
   CrossAccountTransferCapability,
+  CrossAccountWithdrawalRequest,
+  FiatWithdrawalRequest,
   IbanCapability,
   Layer1Cryptocurrency,
-  Layer2Cryptocurrency,
   NationalCurrencyCode,
   PublicBlockchainCapability,
   SwiftCapability,
+  Withdrawal,
   WithdrawalCapability,
+  WithdrawalStatus,
 } from '../../client/generated';
+import { fakeSchemaObject } from '../../schemas';
+import { JSONSchemaFaker } from 'json-schema-faker';
 
-export const WITHDRAWAL_METHODS: WithdrawalCapability[] = [
-  {
-    id: 'e4dee3be-ae75-41d7-83e3-cc29360969b1',
-    balanceAsset: { nationalCurrencyCode: NationalCurrencyCode.USD },
-    withdrawal: {
-      asset: { nationalCurrencyCode: NationalCurrencyCode.USD },
-      transferMethod: IbanCapability.transferMethod.IBAN,
-    },
-  },
-  {
-    id: '1c560908-8704-4706-a732-ab4a54160297',
-    balanceAsset: { assetId: SUPPORTED_ASSETS[0].id },
-    withdrawal: {
-      asset: { cryptocurrencySymbol: Layer1Cryptocurrency.ETH },
-      transferMethod: PublicBlockchainCapability.transferMethod.PUBLIC_BLOCKCHAIN,
-    },
-  },
-  {
-    id: 'fdd88a6d-e9d7-4e30-a495-a8867839f83b',
-    balanceAsset: { cryptocurrencySymbol: Layer2Cryptocurrency.MATIC },
-    withdrawal: {
-      asset: { nationalCurrencyCode: NationalCurrencyCode.MXN },
-      transferMethod: SwiftCapability.transferMethod.SWIFT,
-    },
-  },
-  {
-    id: '4ae61af1-1c23-43ec-8ff6-8892789a266c',
-    balanceAsset: { cryptocurrencySymbol: Layer2Cryptocurrency.MATIC },
-    withdrawal: {
-      asset: { nationalCurrencyCode: NationalCurrencyCode.MXN },
-      transferMethod: CrossAccountTransferCapability.transferMethod.INTERNAL_TRANSFER,
-    },
-  },
-];
+export type WithdrawalRequest =
+  | FiatWithdrawalRequest
+  | BlockchainWithdrawalRequest
+  | CrossAccountWithdrawalRequest;
+
+type Order = 'asc' | 'desc';
+
+export class WithdrawalNotFoundError extends XComError {
+  constructor() {
+    super('Withdrawal not found');
+  }
+}
+
+export class WithdrawalController {
+  private readonly withdrawalRepository = new Repository<Withdrawal>();
+  private readonly withdrawalCapabilityRepository = new Repository<WithdrawalCapability>();
+
+  constructor(
+    assetsConroller: AssetsController,
+    capabilitiesCount: number,
+    withdrawalsCount: number
+  ) {
+    Array.from(Array(capabilitiesCount)).map(() =>
+      this.withdrawalRepository.create(fakeSchemaObject('Withdrawal') as Withdrawal)
+    );
+    Array.from(Array(withdrawalsCount)).map(() =>
+      this.withdrawalCapabilityRepository.create(
+        fakeSchemaObject('WithdrawalCapability') as WithdrawalCapability
+      )
+    );
+
+    this.setKnownAdditionalAssets(assetsConroller);
+  }
+
+  private setKnownAdditionalAssets(assetsController: AssetsController): void {
```

Comment:
This method should be renamed. Took me a lot of time to understand what it does. Also, either remove the argument, or pass `withdrawalRepository` as an argument too.